.github/workflows/main.yml

```diff
@@ -0,0 +1,34 @@
+name: NIDAQmx package
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7"]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install poetry
```

Comment:
There is a setup-poetry action: https://github.com/Gr1N/setup-poetry

We have been successfully using it for measurementlink-python: https://github.com/ni/measurementlink-python/blob/main/.github/workflows/Publish_NIMS.yml#L23